app/models/agents/csv_agent.rb

```diff
@@ -0,0 +1,195 @@
+module Agents
+  class CsvAgent < Agent
+    include FormConfigurable
+    include FileHandling
+
+    cannot_be_scheduled!
+    consumes_file_pointer!
+
+    def default_options
+      {
+        'mode' => 'parse',
+        'separator' => ',',
+        'use_fields' => '',
+        'output' => 'event_per_row',
+        'with_header' => 'true',
+        'data_path' => '$.data',
+        'data_key' => 'data'
+      }
+    end
+
+    description do
+      <<-MD
+        The `CsvAgent` parses or serializes CSV data. When parsing, events can either be emitted for the entire CSV, or one per row.
+
+        Set `mode` to `parse` to parse CSV from incoming event, when set to `serialize` the agent serilizes the data of events to CSV.
+
+        ### Universal options
+
+        Specify the `separator` which is used to seperate the fields from each other (default is `,`).
+
+        `data_key` sets in the key which contains the serialized CSV or parsed CSV data in both the emitted and incoming events
```

Comment:
"data_key sets the key which contains the serialized CSV or parsed CSV data in both the emitted and incoming events"

Isn't this the file pointer, though?